src/Classes/TradeQuery.lua

```diff
@@ -0,0 +1,897 @@
+currencyConversionTradeMap["Blessing of Chayula"] = "blessing-chayula"
+currencyConversionTradeMap["Veiled Chaos Orb"] = "veiled-chaos-orb"
+currencyConversionTradeMap["Enkindling Orb"] = "enkindling-orb"
+currencyConversionTradeMap["Instilling Orb"] = "instilling-orb"
+currencyConversionTradeMap["Sacred Orb"] = "sacred-orb"
+
+
+local TradeQueryClass = newClass("TradeQuery", function(self, itemsTab)
+    self.itemsTab = itemsTab
+	self.itemsTab.leagueDropList = { }
+    -- Note: Per each Check Price button click we do 2 search requests
+	--       Search is the most rate limiting behavior we need to track
+	-- SEARCH REQUEST RATE LIMIT DATA (as of Feb 2021)
+	--	 Up to  5 search requests in a 12 second window
+	--	 Up to 15 search requests in a 62 second window
+	--	 Up to 30 search requests in a 302 second window
+	self.totalPrice = { }
+	self.rate_short_window = { }
+	self.rate_short_time = 12
+	-- we reduce from 5 to 4 since we need 2 search slots for each request
+	self.rate_short_max_searches = 4
+	self.rate_medium_window = { }
+	self.rate_medium_time = 62
+	-- we reduce from 15 to 14 since we need 2 search slots for each request
+	self.rate_medium_max_searches = 14
+	self.rate_long_window = { }
+	self.rate_long_time = 302
+	-- we reduce from 30 to 29 since we need 2 search slots for each request
+	self.rate_long_max_searches = 29
+
+	-- FETCH REQUEST RATE LIMIT DATA (as of Feb 2021)
+	--	 Up to 12 fetch requests in a 6 second window
+	--	 Up to 16 fetch requests in a 14 second window
+	self.rate_short_fetch_window = { }
+	self.rate_short_fetch_time = 6
+	-- we reduce from 12 to 7 since we may want up to 5 fetch slots for each request
+	self.rate_short_max_fetches = 7
+	self.rate_long_fetch_window = { }
+	self.rate_long_fetch_time = 14
+	-- we reduce from 16 to 11 since we may want up to 5 fetch slots for each request
+	self.rate_long_max_fetches = 11
+
+	-- table of price results index by slot and number of fetched results
+	self.resultTbl = { }
+	self.sortedResultTbl = { }
+
+	-- default set of trade item sort selection
+	self.pbSortSelectionIndex = 1
+	self.pbCurrencyConversion = { }
+	self.lastCurrencyConversionRequest = 0
+	self.lastCurrencyFileTime = { }
+	self.pbFileTimestampDiff = { }
+
+    -- set 
+    self.storedGlobalCacheDPSView = GlobalCache.useFullDPS
+	GlobalCache.useFullDPS = GlobalCache.numActiveSkillInFullDPS > 0
+end)
+
+-- Method to process JSON to Lua translation
+function TradeQueryClass:ProcessJSON(json)
+	return dkjson.decode(json)
+end
+
+-- Method to pull down and interpret available leagues from PoE
+function TradeQueryClass:PullLeagueList(controls)
+	local id = LaunchSubScript([[
+		local curl = require("lcurl.safe")
+		local page = ""
+		local easy = curl.easy()
+		easy:setopt{
+			url = "https://api.pathofexile.com/leagues?type=main&compact=1",
+			httpheader = {'Content-Type: application/json', 'Accept: application/json', 'User-Agent: Path of Building/]]..launch.versionNumber..[[ (contact: pob@mailbox.org)'}
```

Comment:
Does `pob@mailbox.org` actually exist or is it just a placeholder?  Might want to change that to the GitHub URL if not.

Reply:
It's what Cinnabarit told me to use.